\documentclass{article}
 
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
%\usepackage[french]{babel}
\usepackage{graphicx}
\usepackage{hyperref}
\usepackage{lmodern}
\usepackage{amsmath}
\usepackage{listings}
\usepackage{enumerate}
\usepackage{amssymb}
\usepackage{amsfonts}
\usepackage{float}

\author{\textbf{Latrille Thibault}, Boyer Hélène, Le Roncé Iris\\
\small Junior Lab ENS Undergraduate Drosophila's genetics Research Program (EUDRP)\\[-0.8ex]
\small \'Ecole Normale Supérieur, Lyon, France\\}
\title{Picture matching adapted to \textit{Drosophila} eyes }

\begin{document}
\maketitle

\begin{center}
\includegraphics[scale=0.3]{logo.jpg} 
\end{center}
\vfill
\begin{abstract}
We report a new software that can be used to accurately analyse planar cell polarity (PCP) defects in \textit{Drosophila} compound eye while combined with a two-color fluorescent imaging system. This system allow visualization of the mosaic adult photoreceptor neurons
(PRs) \textit{in vivo}. Due to physical limitations, the classic visualization method provides image of the cornea that only covers a small part of the eye and the regularity of the eye's shape prevent us to match adjacent images in order to cover the whole eye. Indeed, obtaining a picture of the whole eye is essential to detect some defects such as edge effects that could not be detected with only many part of the eye. We used advantages provided by the visualization method to compute with our software the map of the whole eye by matching adjacent images. PCP features and defects could easily be extracted from this map and statistical analysis is by far improved. Here we test this method on a wild type fly, but it could also be combined with recessive loss-of-function mutation to handle analysis of various genes involves in PCP defects.
\end{abstract}
\newpage
\section{Introduction}
Planar cell polarity (PCP) occurs when the cells of an epithelium are polarized along a common axis lying in the epithelial plane. An example of PCP in Drosophila occurs in the compound eye, where PCP is evident in the orientation of the dorsal and ventral ommatidia, which are mirror images of each other. As well as wings, the eye is a commonly used model to study PCP\cite{ref7}. However many genetics methods has been developed in the eye to study miscellanous mechanistic issues not necessarily related to PCP. We used a new genetic method combined with a dedicated computer tool to improve analysis of eye's PCP issues. Actually, this genetic method termed “Tomato/GFP-FLP/FRT” system provides a two-color in vivo image of the cornea by combining
mitotic recombination and cornea neutralization techniques\cite{ref4}. \\
PCP features of ommatidia units, like dorso–ventral or antero-posterior axis which are defined by the asymmetric organization of the six outer photoreceptor neurons (PRs, R1–R6) \cite{ref4}, can easily get caught by this imaging method.
However, due to the eye curvature, pictures of the eye taken with an optical microscope contain at most 80 ommatidia units among  a total number of $\simeq 800$ ommatidia. That issue could limit detection of PCP defect or statistical analysis. We overcame this limitation by designing an experimental set-up allowing us to spin the fly below the microscope's lens. Then we matched adjacent images using the irregular eye colour pattern generated by the “Tomato/GFP-FLP/FRT” system. Indeed, While the red dye could be stochastically expressed in photoreceptor neurons, it broke the regularity of the pattern of the compound eye extracted with the green dye. Using Python language, we designed a dedicated matching software which we named Flymatch toolbox. This program can be easily mastered through it Graphics User Interface (GUI) and provides both accurate and efficient way to build up Drosophila eye map facilitating analysis. General statistical analyses of the map is not yet implemented in our software but one could easily code specific analysis based on the final data.

\newpage
\section{Methods and results}


\subsection{The Tomato/GFP-FLP/FRT system}
This techniques combine mitotic recombination and the classic UAS-Gal4 system to generate two-colour clone in the \textit{Drosophila} eye.
We first mated $rh1$-$Gal4$, $ey$-$FLP$ ; $FRT42D$, $rh1$-$tdTomato^{ninaC}$/$CyO$ ; $UAS$-$GFP^{ninaC}$ virgin females with $FRT42$/$CyO$ males to get the required genetic background (Fig. 1), these two lines are a kind gift of Bertrand Mollereau.
\begin{figure}[H]
	  \centering
  	\includegraphics[width=0.9\textwidth]{X.png}
  	\caption{\textbf{The Tomato/GFP-FLP/FRT system.} \textsl{F1 flies carry on the GFP marker under the control of rh1 (a driver specifically expressed in R1-6 photoreceptors), and rh1-tdTomato construct (red marker) onto a FRT chromosome. The F1 is heterozygous for both construct. FLP-FRT mediated mitotic recombinations generate clones of homozygous $FRT$ cells (green) in a heterozygous and homozygous $FRT$-$tdTomato^{ninaC}$ background (yellow).}
  	}
	\end{figure}

Fly resulting from the crossbreed are heterozygous for two flurophores. PRs expressing GFP under the rhodopsin 1 (rh1) GAL4 driver are directly visualized through the cornea in anesthetized and water-covered flies with an immersion objective (Fig. 2A). In this background, we introduced the tdTomato fluorescent marker \cite{ref3} under the control of rh1 promoter (rh1-tdTomato) onto an FRT chromosome. After FLP-FRT-mediated mitotic recombination, only a part of the PRs population expresses the tdTomato marker in a background where all PRs express GFP\cite{ref}\cite{ref1} (Fig. 2B). This leads to different PR populations distinguishable in mosaic analysis by
the cornea neutralization technique\cite{ref2}. Pattern of the mutant clones varied from large clones to more scattered ones. This opens new opportunities to match adjacent images, breaking down the regularity of the GFP background. In fact, a regular pattern prevents matching since all images are nearly similar.
\begin{figure}[H]
	  \centering
  	\includegraphics[width=1\textwidth]{merge.png}
  	\caption{\textbf{Mosaic expression of tdTomato in the retina.
} \textsl{Random recombination with FRT-FLP generate two-color clones. Only PRs with no recombination or homozygous $FRT-tdTomato^{ninaC}$ cells carry the tomato marker (B) in a GFP background (A). The stochastic red pattern bring sufficient irregularity to match adjacent images.}}
	\end{figure}


\subsection{Rotate the fly below the lens}
Sweeping the whole eye could be done only by twirling the fly below the microscope. Indeed increasing resolution with confocal microscopy is not sufficient to obtain all PRs from a single point of view since PRs with a too much large angle between the direction of the ommatidium and the lens axis are hidden by pigment cell surrounding the rhabdomeres (PRs).
To overcome this difficulty we created a two degrees of freedom experimental set-up to spin the fly's eye. Briefly, flies are fixed by the body onto the mobile part of the device. While imaging, the mobile part is manually moved to get images of the eye from different angles, to viusalize the whole compound eye.

	\begin{figure}[H]
	  \centering
  	\includegraphics[width=0.6\textwidth]{montage.jpg}
  	\caption{\textbf{Experimental set-up for whole Drosophila imaging.} \textsl{The set-up is composed of a fixed platform (in black) and a mobile piece (in red) with two degrees of freedom (orange and blue arrows). The fly's eye is covered with water which neutralizes refraction indices of the cornea. Due to surface tension, a single drop of water could hold between lens and mobile part, which is sufficient to neutralize the refraction.}}
	\end{figure}

\subsection{Flymatch toolbox}
Using our new acquisition device we generated a huge amount of images. Thus, we designed a software to annotate, edit and assemble these image and generate a 3D reproduction of the eye. Computing the 3D representation necessitate two major steps, the first one is the extraction of relevant data and the second one is the mapping and assembly of this data.
To extract data (first step), the user needs to click all PRs according to the accepted numbering, that is to say his first click is the PR number one, the second click is PR number two, up to the sixth PR which complete an ommatidium. Then the user can click an other ommatidium, complete it and so on. There is only six PRs in our ommmatidia since R7 and R8 are not driven by the rh1 promoter and thus are not coloured.
Our software auto-detects the PR's color and the ommatidium orientation (Fig. 4).
\begin{figure}[H]
	  \centering
  	\includegraphics[width=1\textwidth]{live.png}
  	\caption{\textbf{Screen-shot of the program during execution.} \textsl{Yellow and green dots are PRs edited by the user, this information is used by the computer to realise the matching. Black arrows represent ommatidia's orientation. Boxes to the right are used to help the user (display and matching process).}}
	\end{figure}

Once all images are edited, matching process (second step) can be done. Matching begin by the projection of extracted data onto the unit sphere. Indeed it is necessary to map two-dimensional coordinate onto a sphere (see appendix) if we want to reproduce the curvature of the eye. Then the matching process is realised according to the sequence described below where the error function is a function that compute the mismatch rate (see next section).
\begin{enumerate}
		\item{Entry :\{$S_{1}$,$S_{2}$\},Two sets of ommatidia mapped onto the sphere.}
		\item{For each possible pair {(p,q) such as $p\in S_{1},q\in S_{2}$}: }
			\begin{enumerate}
			\item{move $S_{2}$ in a way p match with q (Fig. 5)}
			\item{compute the error function on moved $S_{1}$ and $S_{2}$}
			\end{enumerate}
		\item{Select the best match (error function minimum)}
		\item{Compute $S_{1}\cup S_{2}$ of the best match}
	\end{enumerate}

Matching process is implemented in two different modes, an user-guided mode and an auto-guided
one. The first is labouring for the user but allow a more fast and accurate
matching whereas the last could be very long in term of computing time. The user-guided mode uses the same procedure but the program iterates only among pair defined by the user.

\begin{figure}[H]
	  \centering
  	\includegraphics[width=1\textwidth]{move.png}
  	\caption{\textbf{Image matching method to obtain a 3D view.} \textsl{The coloured axis represent the x (white), y (red) and z (blue) axis. Let $S_{1}$, $S_{2}$ be sets of ommatidia.
  	 Let i $\in$ $S_{1}$ and j $\in$ $S_{2}$ (1).
		Move i to the south pole (2).
		Move j to the south pole (3).
		Spin $S_{2}$ around the south pole.
		Thus i and j are in perfect match (4) and now the program can compute the error function. 
		The south pole is a convenient arbitrary choice.
		This steps are realised for each possible couple {(i,j) such as $i\in S_{1},j\in S_{2}$} and the best match is obtained for error function minimum.
		}}
	\end{figure}

We used quaternions to fasten the application. Briefly, quaternions are an extension of imaginary numbers in a four-dimensional vector space over the real numbers. Any rotation in three-dimensions can be represented as an axis vector and an angle of rotation. 
Quaternions give a simple way to encode this axis-angle representation in four numbers and apply the corresponding rotation to position vectors representing points relative to the origin\cite{ref6}. 
As the main loop (second step) of our program induces repetitive rotation of $S_{1}$ and $S_{2}$ around axis (Fig. 5) we used quaternions to makes our application runs significantly faster, it is also easy to implement.\\
We supposed in this method the eye is a perfect hemisphere, obviously it is not. However our final data proved that in first approximation the eye is not far away from a prefect hemisphere. 

\subsection{Error function}
The error function is applied to two moved sets of ommatidia after a couple of ommatidia is aligned.\\
		Let $\displaystyle (\alpha, \beta, \gamma) \in \mathbb{R}_{+*}^3$ be a set of constants determined empirically to weighted the different terms.\\
		Let $ S_{1} $, $ S_{2} $ be two sets of ommatidia with a determined position onto the sphere.\\
		Let $m$ be the number of pairs of ommatidia. A pair of ommatidia is defined as a couple $(i,j)$ of ommatidia such as $i \in S_{1}$, $j \in S_{2}$ and $j$ is the corresponding (matching) ommatidia of $i$ in $ S_{2} $. 
		\\
		Let $D\in \mathbb{R}_{+*}$ a threshold determined empirically, used to know if a couple of ommatidia is forming a pair.
	\begin{center}
       $\displaystyle F(S_{1},S_{2})=\sum_{i \in S_{1}, j \in S_{2}} \psi(i,j)*m^{-2}*\left[\alpha d(i,j)+\beta \sum_{k\in\lbrace1,...,6\rbrace}\delta_{ijk}+\gamma (1-<i,j>)^2 \right] $
     \end{center}
     \begin{center}
     	$\displaystyle d(i,j)$ the euclidian distance between $i$ and $j$.
     	\\ $\psi(i,j)=1$ if $\displaystyle d(i,j)<D$, $0$ otherwise. That is to say only ommatidia with close neighbours (paired) are ultimately relevant.
       \\ Let $P_{ik}$ ($P_{jk}$) be the color of photoreceptor $k$ in ommatidium$i$ ($j$).
       \\$\displaystyle \delta_{ijk}=1 $ if $P_{ik} \neq P_{jk}$, $0$ otherwise.
       \\$\displaystyle <i,j>$ the scalar product between orientation of $i$ and $j$.
	\end{center}
Let us explain the meaning of each term.\\
Close ommatidia are preferred, inducing the term $ d(i,j) $.
\\
The function $ \sum_{k\in\lbrace1,...,6\rbrace}\delta_{ijk}  $ is explained by the preference for ommatidia owning PRs with the same color.
 \\
Ommatidia with the same orientation are preferred, inducing the term $ (1-<i,j>)^2 $.
\\
 Matching with many ommatidia are preferred than those with only a few, inducing the term $ m^{-2} $.
 \\
 Thus F increase as the badness of a match do.
 \\
 Regarding to indices below the sum, one could naively think computing the error function have complexity $O(n^{2})$, in fact computing the error function have complexity $O(nln(n))$ using KD-Tree. Indeed, we first iterate over the elements in $ S_{1} $, then we search the elements of $S_{2}$ in the KD-tree below the threshold. Searching into the KD-tree have asymptotic complexity $O(ln(n))$. \\
 As we computed the error function for each aligned couple, the overall program found the best match in $O(n^{3}ln(n))$.

\subsection{Results \& Mutant analysis}
For the main experiment, we used 45 images taken from a single fly (Fig. 6). Though, the final data covered $\simeq 60 \% $ of the eye. We expect to cover the whole eye with a gathering of $\simeq 60$ images, but we were not able to do so because bleaching still remains a major issue and limits the exposition time. One could overcome this limitation using a confocal microscope. 
\begin{figure}[H]
	  \centering
  	\includegraphics[width=0.95\textwidth]{compil.png}
  	\caption{\textbf{Sequential image assembly for the 3D view.} \textsl{We successfully matched forty-five images (4), beginning by a match of two images (1). Matching 10 images (2) and 25 images (3) are shown. However bleaching is a major concern we need to overcome to improve our method.}}
	\end{figure}
A common way to asses the influence of a gene on PCP is to compare a wild type fly to a one with a mutated gene (KO-gene), however our method provide a more elegant way to do so. We mated $rh1$-$Gal4$, $ey$-$FLP$ ; $FRT42D$, $rh1$-$tdTomato^{ninaC}$/$CyO$ ; $UAS$-$GFP^{ninaC}$ virgin females with FRT42, KO-gene /CyO males.
\\ After FLP-FRT-mediated mitotic recombination the mosaic is composed of three population :
\begin{enumerate}
		\item{Homozygous $ \dfrac{FRT42, KO-gene}{FRT42, KO-gene} $, only green coloured.}
		\item{Homozygous $ \dfrac{FRT42, rh1-tdTomato^{ninaC}}{FRT42, rh1-tdTomato^{ninaC}} $, green and red coloured.}
		\item{Heterozygous $ \dfrac{FRT42 KO-gene}{FRT42, rh1-tdTomato^{ninaC}} $, green and red coloured.}
\end{enumerate}
 Thus, when studying recessive mutations, homozygous green mutant PRs will exhibit mutant phenotype in a background composed of wild-type yellow PRs. So we could compare the effect of the mutation on a single fly (Fig. 7).
\begin{figure}[H]
	  \centering
  	\includegraphics[width=0.8\textwidth]{fluo.png}
  	\caption{\textbf{Apply the Flymatch toolbox to analyse PCP mutants.} \textsl{Our method could handle analysis of PCP defects in a mutate background. Green PRs are homozygous for the mutation, yellow PRs are either Wild-Type or heterozygous for the mutation. Here, the mutation is a recessive loss of function of the transcription co-activator mastermind.}}
	\end{figure}
	

\section{Discussion \& perspectives}
Our method, by combining visualization of the whole compound eye and computing the map of the eye, actually yields a list of coordinates of ommatidia and his PRs. Thus the direction of the ommatidia could be easily computed. Assemble all orientations and one get the map of ommatidia's orientation, this maps contains all PCP features and thus can be used to detect PCP defects.
However the process is labouring and one could preferred not to annotate images. Some softwares are able to extract automatically features using \textit{Scale-invariant feature transform} \cite{ref5}. However one needs to specifically adapt this method to his data because treatment depends on which features you want to extract, additionally this technique can be subjected to errors. \\
\\Moreover bleaching is a major concern that limit the exposition time, thus the accuracy of the method. \\ 
To summarize, combining a novel genetic method and informatics program, we successfully obtain a map of the \textit{Drosphila}'s eye containing PCP features. This method could be associated with PCP defects analysis(Fig. 7), with a more accurate fashion than ever before. Our method could also be adapted to obtain a full photographic coverage of an object like wings, body, etc if this object is sufficiently irregular (see Fig. 2). However, to be convenient and widely used, our method needs many improvements and works (Fig. 6)


\section{Acknowledgements}
We are grateful to Bertrand Mollereau's team for continuous discussion and enjoyment.
We thank Pauline Marangoni for her care to us and her support.
We also wish to thank Bertrand Mollereau for settling-up the project and his belief that undergraduate students could experience their own research project.
And we thank all coordinators that allowed the Junior Lab to be set-up.
We also would like thank the DGRC and the Bloomington stock center for flies, as well as Drosotool facilities and Florence Lormieres, Laurence Belgari-Dutron and Fanny Chaumeille-Dole for technical help. 
We really hope future students to experience a project like this one.

\appendix
	
\section{Projecting the plan onto a sphere}
We used stereographic projection since it is conformal, meaning it preserve angles. However any map from the sphere to the plane can be both conformal and area-preserving. \\
In Cartesian coordinates (x, y, z) on the sphere and (X, Y) on the plane, the stereographic projection and its inverse are given by the formulas :\\
\begin{flushleft}
$\displaystyle (X,Y)=(\dfrac{x}{1-z},\dfrac{y}{1-z})$\\
$\displaystyle (x,y,z)=(\dfrac{2X}{1+X^2+Y^2},\dfrac{2Y}{1+X^2+Y^2},\dfrac{X^2+Y^2-1}{1+X^2+Y^2})$\\
\end{flushleft}
Before being projected, raw coordinate extracted in pixels are scaled by a factor taking into account the image size, the eye's size and the optical and numerical devices. 

\bibliographystyle{unsrt} % Le style est mis entre crochets.
\bibliography{bibli}

\end{document}
